pinot-controller/src/main/java/org/apache/pinot/controller/recommender/io/InputManager.java

```diff
@@ -237,11 +237,16 @@ private void reorderDimsAndBuildMap()
     _colNameToIntMap = new HashMap<>();
 
     _dimNamesInvertedSortedIndexApplicable = new HashSet<>(_dimNames);
+    _dimNamesInvertedSortedIndexApplicable.addAll(_metricNames);
```

Comment:
(nit) add a comment highlighting that all types of comments will be considered by the rule

Reply:
Also suggest renaming this to `_columnNamesInvertedSortedIndexApplicable` since this PR removes the restriction that it has to be a dimension

Reply:
refactored at other places as well!